RuntimeError: Broken installation: can't determine base dir

When installing via portage (I'm working on packaging mypy for portage), I get the following when trying to run mypy:

```
Traceback (most recent call last):
  File "/usr/lib/python-exec/python3.3/mypy", line 169, in <module>
    main()
  File "/usr/lib/python-exec/python3.3/mypy", line 34, in main
    type_check_only(path, module, bin_dir, options)
  File "/usr/lib/python-exec/python3.3/mypy", line 78, in type_check_only
    python_path=options.python_path)
  File "/usr/lib64/python3.3/site-packages/mypy/build.py", line 122, in build
    data_dir = default_data_dir(bin_dir)
  File "/usr/lib64/python3.3/site-packages/mypy/build.py", line 191, in default_data_dir
    raise RuntimeError("Broken installation: can't determine base dir")
RuntimeError: Broken installation: can't determine base dir
```

I'm not sure what directory it's starting with but I can tell it's not finding what it's looking for.  Any guidance would be appreciated.

My environment is as below. `mypy` works with some fixes.

* Windows 7/32bit
* Python 3.5
* mypy version: latest (`pip install git+https://github.com/JukkaL/mypy.git`)
* running on Git Shell (msys2)

There are some points.

* I use latest mypy (`master` branch) because mypy on PyPI doesn't support Python 3.5 (PyPI version causes `ImportError: cannot import name 'Undefined'` due to the removing of `Undefined` type (please refer #639)).
* I have to modify [`mypy` script file's header](https://github.com/JukkaL/mypy/blob/master/scripts/mypy#L1) from `#!/usr/bin/env python3` to `#!/usr/bin/env python` because that Windows doesn't have `python3` command (If you use command prompt, it doesn't matter).

And the `builtins.pyi` is located in `(my virtualenv folder)\Lib\mypy\typeshed\builtins\3`.

Then `mypy` works.

```
$ mypy exmaple.py
exmaple.py:5: error: Argument 1 to "sum" has incompatible type "float"; expected "int"
```

`example.py` is like below.

```
def sum(x: int, y:int) -> int:
    return x + y

if __name__ == "__main__":
    print(sum(0.1, 2))  # causes type check error
```

--- a/mypy/build.py
+++ b/mypy/build.py
@@ -285,6 +285,9 @@ def default_data_dir(bin_dir: str) -> str:
     elif base == 'python3':
         # Assume we installed python3 with brew on os x
         return os.path.join(os.path.dirname(dir), 'lib', 'mypy')
+    elif dir.endswith('python-exec'):
+        # Gentoo uses a python wrapper in /usr/lib to which mypy is a symlink.
+        return os.path.join(os.path.dirname(dir), 'mypy')
     else:
         # Don't know where to find the data files!
         raise RuntimeError("Broken installation: can't determine base dir")

